src/main/java/com/ericsson/ei/handlers/MongoDBHandler.java

```diff
@@ -266,26 +268,45 @@ public void createTTLIndex(String dataBaseName, String collectionName, String fi
     private MongoCollection<Document> getMongoCollection(String dataBaseName, String collectionName) {
         if (mongoClient == null)
             return null;
-        MongoDatabase db = mongoClient.getDatabase(dataBaseName);
-        List<String> collectionList = db.listCollectionNames().into(new ArrayList<String>());
-        if (!collectionList.contains(collectionName)) {
-            LOGGER.debug("The requested database({}) / collection({}) not available in mongodb, Creating ........", dataBaseName, collectionName);
-            try {
-                db.createCollection(collectionName);
-            } catch (MongoCommandException e) {
-                String message = "collection '" + dataBaseName + "." + collectionName + "' already exists";
-                if (e.getMessage().contains(message)) {
-                    LOGGER.warn("A {}.", message, e);
-                } else {
-                    throw e;
-                }
-            }
-            LOGGER.debug("done....");
+        MongoDatabase db;
+        List<String> collectionList;
+        MongoCollection<Document> collection = null;
+        try {
+            db = mongoClient.getDatabase(dataBaseName);
+            collectionList = db.listCollectionNames().into(new ArrayList<String>());
+            if (!collectionList.contains(collectionName)) {           
+                LOGGER.debug("The requested database({}) / collection({}) not available in mongodb, Creating ........", dataBaseName, collectionName);
+                db.createCollection(collectionName);                
+            }    
+            collection = db.getCollection(collectionName);            
         }
-        MongoCollection<Document> collection = db.getCollection(collectionName);
+        catch (MongoCommandException e) {
+            LOGGER.error("MongoCommandException, Something went wrong with MongoDb connection. Error: " + e.getErrorMessage() + "\nStacktrace\n" + e);
+            closeMongoDbConnection();
```

Comment:
"e" doesnt include stacktrace. You will have to use ExceptionUtils.getStacktrace() in order to get it.

Reply:
The idea is not to include full stacktrace to Log.Debug message.
Maybe we should add all stacktraces to LOGGER.trace(...) or something. So we can control stacktrace in logs for Java properties. So we can steer/control the amount of logs. Logging stacktrace discussion and implementation should be an own story/task.

Reply:
But stacktrace in Logger.error is fully okay is it not? Also, you are writing "\nStacktrace\n" but it's not a stacktrace. Only "e" is not saying much at all.

Reply:
That is what the new story/staks should discuss and implement the decided solution.
I think it should be possible to turn of stacktraces, for example reducing log printouts in Travis CI.

Reply:
If you put stacktraces in LOGGER.error, you can just set the logging level to debug or something in travis if you don't want the stacktraces. But yes, we should not keep stacktraces in Log.debug messages.

Reply:
Okay, just remove the "\nStacktrace\n" + e part then :)

Reply:
fixed.